models/badges.js

```diff
@@ -27,6 +27,35 @@ const fetchBadges = async ({
   }
 }
 
+/**
+ * Fetches the data about user badges
+ * @param query { Object }: Filter for badges data
+ * @return {Promise<userBadgeModel|Array>}
+ */
+
+const fetchUserBadges = async (username) => {
+  try {
+    const snapshot = await badgeModel.get()
+    const allBadges = []
+    snapshot.forEach((doc) => {
+      allBadges.push(doc.data())
+    })
+    const userBadges = []
+    allBadges.forEach((badge) => {
+      badge.users.forEach((user) => {
+        if (user === username) {
```

Comment:
Are we not going to use userId? What if someone changes their username? This API won't work for them
Check functions defined in utils/users for the conversion

Reply:
The requirements specified that we use the username of the member 😅

Reply:
It would be better to go with unique userid in my opinion, we can discuss on that

Reply:
It would be better to go with unique userid in my opinion, we can discuss on that